BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/ble/beacon/signal/KalmanFilter.java

```diff
@@ -39,25 +38,22 @@
      */
     private static float DEFAULT_MEASUREMENT_NOISE = 10;
 
-    private double processNoise;
-    private double measurementNoise;
-    private double estimatedRSSI;
-    private double errorCovarianceRSSI;
+    private float processNoise = DEFAULT_PROCESS_NOISE;
+    private float measurementNoise = DEFAULT_MEASUREMENT_NOISE;
+    private float estimatedRssi;
+    private float errorCovarianceRssi;
     private boolean isInitialized = false;
 
-    public KalmanFilter(long duration, TimeUnit timeUnit) {
-        super(duration,timeUnit);
-        this.processNoise = DEFAULT_PROCESS_NOISE;
-        this.measurementNoise = DEFAULT_MEASUREMENT_NOISE;
+    public KalmanFilter() {
```

Comment:
You will not be able to create an `KalmanFilter` with custom `duration`, `timeUnit`, `processNoise` and `measurementNoise` on construction.